novawallet/Modules/NetworksList/NetworksListViewController.swift

```diff
@@ -0,0 +1,230 @@
+import UIKit
+import SoraFoundation
+
+final class NetworksListViewController: UIViewController, ViewHolder {
+    typealias RootViewType = NetworksListViewLayout
+    typealias ViewModel = RootViewType.Model
+    typealias DataSource = UITableViewDiffableDataSource<RootViewType.Section, RootViewType.Row>
+    typealias Snapshot = NSDiffableDataSourceSnapshot<RootViewType.Section, RootViewType.Row>
+    typealias ChainCell = NetworksListTableViewCell
+
+    let presenter: NetworksListPresenterProtocol
+
+    private var viewModel: ViewModel?
+    private var networkViewModels: [RootViewType.NetworkWithConnectionModel] = []
+
+    private lazy var dataSource = makeDataSource()
+
+    init(
+        presenter: NetworksListPresenterProtocol,
+        localizationManager: LocalizationManagerProtocol
+    ) {
+        self.presenter = presenter
+        super.init(nibName: nil, bundle: nil)
+        self.localizationManager = localizationManager
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func loadView() {
+        view = NetworksListViewLayout()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setup()
+        presenter.setup()
+    }
+}
+
+// MARK: UITableViewDelegate
+
+extension NetworksListViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        presenter.selectChain(at: indexPath.row)
+    }
+
+    func tableView(_: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        guard let section = viewModel?.sections[indexPath.section] else { return .zero }
+
+        if case .networks = section {
+            return 56
+        } else {
+            return UITableView.automaticDimension
+        }
+    }
+}
+
+// MARK: NetworksListViewProtocol
+
+extension NetworksListViewController: NetworksListViewProtocol {
+    func update(with viewModel: NetworksListViewLayout.Model) {
+        self.viewModel = viewModel
+
+        var snapshot = Snapshot()
+
+        viewModel.sections.forEach { section in
+            snapshot.appendSections([section])
+
+            switch section {
+            case let .networks(rows), let .banner(rows):
+                snapshot.appendItems(rows)
+
+                networkViewModels = rows.compactMap { row in
```

Comment:
I think we can just switch to manual delegate and provide cell because snapshotting only complicates things if we have a separate data source